### PR TITLE
In pbft keep tracks of the last pre-prepare that performed a governance request

### PR DESCRIPTION
**Pull request by ashamis**
related to #953

This change creates a "chain" of pre-prepares that contain requests that performed some kind of governance operation.  What this means that when a user gets their receipt (which covers a pre-prepare) they can make a determination if they want to audit the log and if they do they do not need to look into every pre-prepare.

### CODE REVIEW

File: src/consensus/pbft/libbyz/Replica.h [hunk 1]
```diff
@@ -497,6 +499,10 @@ class Replica : public Node, public IMessageReceiveBase
   // used to register a callback for a client proxy to collect replies sent to
   // this replica
 
+  GovernanceRequestTracking gov_req_track;
+  // used to track the last series of pre-prepares that had requests that
+  // affected the goverenance of the service
```
Comment (jumaffre) [inline src/consensus/pbft/libbyz/Replica.h]: governance

File: src/consensus/pbft/libbyz/Pre_prepare.cpp [hunk 2]
```diff
@@ -34,6 +34,8 @@ Pre_prepare::Pre_prepare(
   rep().view = v;
```
Comment (jumaffre) [inline src/consensus/pbft/libbyz/Pre_prepare.cpp]: Why is this change necessary? Ideally, the consensus should be agnostic to the type of transaction being executed?

File: src/consensus/pbft/libbyz/Replica.h (hunk 1 above)
Comment (ashamis) [reply src/consensus/pbft/libbyz/Replica.h]: thanks! my spell checking adding is really failing me lately. Any chance you have a good one?

File: src/consensus/pbft/libbyz/Pre_prepare.cpp (hunk 2 above)
Comment (achamayou) [reply src/consensus/pbft/libbyz/Pre_prepare.cpp]: I'm also curious.

File: src/consensus/pbft/libbyz/Pre_prepare.h [hunk 3]
```diff
@@ -35,6 +35,8 @@ struct Pre_prepare_rep : public Message_rep
   View view;
   Seqno seqno;
   std::array<uint8_t, MERKLE_ROOT_SIZE> replicated_state_merkle_root;
+  uint64_t contains_gov_req;
```
Comment (achamayou) [inline src/consensus/pbft/libbyz/Pre_prepare.h]: Why not bool?

File: src/node/rpc/frontend.h [hunk 4]
```diff
@@ -653,5 +653,10 @@ namespace ccf
     {
       return true;
     }
+
+    virtual bool is_governance_front_end() override
```
Comment (achamayou) [inline src/node/rpc/frontend.h]: Members frontend?

File: src/consensus/pbft/libbyz/Pre_prepare.cpp (hunk 2 above)
Comment (ashamis) [reply src/consensus/pbft/libbyz/Pre_prepare.cpp]: spoke off-line and updated the comment

File: src/consensus/pbft/libbyz/Pre_prepare.h (hunk 3 above)
Comment (ashamis) [reply src/consensus/pbft/libbyz/Pre_prepare.h]: for alignment purposes :(

Comment (achamayou) [reply src/consensus/pbft/libbyz/Pre_prepare.h]: Maybe good to have a comment then ;)
